consensus/ethash/ethash.go

```diff
@@ -223,7 +223,7 @@ func (c *cache) generate(dir string, limit int, test bool) {
 			size = 1024
 		}
 		// If we don't store anything on disk, generate and return.
-		if dir == "" {
+		if dir == "" || limit <= 0 {
```

Comment:
The logic of `func (c *cache) generate(dir string, limit int, test bool)` has already guaranteed that cache is generated before use. This method would firstly load the cache file and memory map it. If the cache doesn't exist, it would create a new one and memory map it. So it seems that the mechanism here has no problem.

The parameter `limit` here indicate that how many cache files need to exist on disk. Even if `limit <= 0` , it doesn't directly indicate that no cache file exists. At least using `limit <= 0` here is not rigorous logically.

Reply:
The intention here is to reuse the limit flag so that there is a way to disable `mmap`, which is impossible for existing logic. Of course we can introduce a new flag but I believe implicitly disabling `mmap` when limit is not a positive number is enough.